### PR TITLE
udocker 1.0.3 and dependencies

### PR DESCRIPTION
**Pull request by pvanheus**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [x] This PR adds a new recipe.
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

This add `udocker` 1.0.3, its depedency `proot` and proot's dependency `talloc`. udocker (https://github.com/indigo-dc/udocker) is a "basic user tool to execute simple docker containers in batch or interactive systems without root privileges".

### CODE REVIEW

File: recipes/talloc/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  license: LGPL-3.0
+  summary: 'talloc is a hierarchical, reference counted memory pool system with destructors.'
+
+build:
+  number: 0
+  skip: True # [not linux]
+
+requirements:
+  build:
+    - python <3
```
Comment (bgruening) [inline recipes/talloc/meta.yaml]: can you remove the pinning here and skip the build in the build section?

File: recipes/udocker/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  license_file: LICENSE
+  summary: 'Freely available tools for computational molecular biology.'
+
+build:
+  number: 0
+  skip: True # [not linux]
+
+requirements:
+  build:
+    - python <3
```
Comment (bgruening) [inline recipes/udocker/meta.yaml]: same here, please skip above.

File: recipes/talloc/meta.yaml (hunk 1 above)
Comment (bgruening) [inline recipes/talloc/meta.yaml]: ping @pvanheus

Comment (pvanheus) [inline recipes/talloc/meta.yaml]: done... see commit 8e6ead0

**Pull request merged**